pkg/readiness/ready_tracker.go

```diff
@@ -382,8 +398,18 @@ func (t *Tracker) Run(ctx context.Context) error {
 		}
 	})
 
-	_ = grp.Wait()
-	_ = t.constraintTrackers.Wait() // Must appear after grp.Wait() - allows trackConstraintTemplates() time to schedule its sub-tasks.
```

Comment:
nit: might be good to restore this comment for future reference.

Reply:
I don't think the comment is exactly accurate anymore, but I can add something similar to explain why it comes after the grp.Wait()